ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -288,29 +294,48 @@ localRootPeersProvider tracer timeout resolvConf rootPeersGroupsVar domainsGroup
 #else
     let rr = newResolverResource resolvConf
 #endif
-    -- Flatten the local root peers groups and associate a group index to each
-    -- DomainAddress to be monitorized.
     let domainsGroupsList = NonEmpty.toList domainsGroups
+
+        -- Flatten the local root peers groups and associate a group index to each
+        -- DomainAddress to be monitorized.
+        addresses = [ (group, address, pa)
+                    | (group, (_, m)) <- zip [0..] domainsGroupsList
+                    , (address, pa) <- Map.toList m ]
         domains = [ (group, domain, pa)
-                  | (group, (_, m)) <- zip [0..] domainsGroupsList
-                  , (domain, pa) <- Map.toList m ]
-    -- Since we want to preserve the number of groups, the targets, and
-    -- the addresses within each group, we fill the TVar with
-    -- a placeholder list, in order for each monitored DomainAddress to
-    -- be updated in the correct group.
+                  | (group, RelayDomain domain, pa) <- addresses ]
+        ipsAddr = [ (group, IP.toSockAddr (ip, port), pa)
+                  | (group, RelayAddress ip port, pa) <- addresses ]
+
+        -- Since we want to preserve the number of groups, the targets, and
+        -- the addresses within each group, we fill the TVar with
+        -- a placeholder list, in order for each monitored DomainAddress to
+        -- be updated in the correct group.
+        placeholder = map (\(t, _) -> (t, Map.empty)) domainsGroupsList
```

Comment:
please add type annotations for `addresses`, `domains` and `placeholder`.  I find it much easier to read the code when I don't need to infer the types.